packages/core-mobile/app/services/activity/ActivityService.tsx

```diff
@@ -30,6 +40,58 @@ export class ActivityService {
       nextPageToken: rawTxHistory.nextPageToken
     }
   }
+
+  /**
+   * Enriches SVM networks with SPL token metadata for proper transaction history display.
+   *
+   * Unlike EVM networks which include token metadata in the main /networks endpoint,
+   * Solana networks return empty tokens arrays and require fetching from /tokenlist?includeSolana.
+   * This ensures SPL tokens show proper symbols (e.g., "ORCA") instead of "Unknown" in activity.
+   */
+  private async enrichNetworkWithTokens(network: Network): Promise<Network> {
+    // Only enrich SVM networks that don't already have tokens
+    if (
+      network.vmName !== NetworkVMType.SVM ||
+      (network.tokens && network.tokens.length > 0)
+    ) {
+      return network
+    }
+
+    // Try to get cached data first
+    const queryKey = ['tokenList', 'solana', network.chainId]
+    const cachedData =
+      queryClient.getQueryData<NetworkContractToken[]>(queryKey)
+
+    if (cachedData) {
+      return {
+        ...network,
+        tokens: cachedData
+      }
+    }
+
+    // If no cached data, fetch and cache
+    try {
+      const tokenListResponse = await fetch(
+        `${Config.PROXY_URL}/tokenlist?includeSolana`
+      )
+      const tokenData = await tokenListResponse.json()
```

Comment:
looks like the query is never executed since fetch is still being used. you can do something like this:
```
export const getCachedTokenList = async (chainId: string) => {
  return queryClient.fetchQuery({
    queryKey: ['tokenList', 'solana'],
    queryFn: fetchTokenListWithSolana,
    staleTime,
    gcTime,
    select: (data) => data[chainId]?.tokens ?? []
  })
}

getCachedTokenList()
```